modules/nf-core/gsea/gsea/main.nf

```diff
@@ -43,43 +43,41 @@ process GSEA_GSEA {
     def VERSION = '4.3.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+
+    // Remove any trailing dots from prefix when passed as report label, so
+    // GSEA doesn't produce double-dotted top-level outputs
+    def rpt_label = prefix.replaceAll(".\$", "")
```

Comment:
Shouldn't the pattern be "\.*$" to match trailing dots? This one looks like you're removing any character followed by a dollar.

Reply:
Good catch- thanks. 

It actually needs to be `\\.$` to match a single trailing dot (`.` is interpreted as any character so would actually strip a single trailing character from all prefixes, which isn't what I was going for!). The escaping on the dollar was down to my double quotes:

```
> println('foo.bar.'.replaceAll(".", ''))
// (empty string)

> println('foo.bar.'.replaceAll("\\.", ''))
foobar

> println('foo.bar.'.replaceAll("\\.$", ''))
unknown recognition error type: groovyjarjarantlr4.v4.runtime.LexerNoViableAltException

> println('foo.bar.'.replaceAll('\\.$', ''))
foo.bar

> println('foo.bar'.replaceAll('\\.$', ''))
foo.bar
```